platform/viewer/src/components/ViewportGrid.tsx

```diff
@@ -42,18 +42,23 @@ function ViewerViewportGrid(props) {
           continue;
         }
 
+        // console.log("Applying HP matchDetails", i, matchDetails[i]);
         // if current viewport doesn't have a match
         if (matchDetails[i] === undefined) return;
 
         const { displaySetsInfo, viewportOptions } = matchDetails[i];
 
         const displaySetUIDsToHang = [];
         const displaySetUIDsToHangOptions = [];
-        displaySetsInfo.forEach(({ SeriesInstanceUID, displaySetOptions }) => {
-          const matchingDisplaySet = availableDisplaySets.find(ds => {
-            return ds.SeriesInstanceUID === SeriesInstanceUID;
-          });
-
+        displaySetsInfo.forEach(({ SeriesInstanceUID, displaySetInstanceUID, displaySetOptions }) => {
+          const matchingDisplaySet =
+            availableDisplaySets.find(ds => ds.displaySetInstanceUID == displaySetInstanceUID) ||
+            availableDisplaySets.find(ds => ds.SeriesInstanceUID === SeriesInstanceUID);
+
+          // console.log("Displaying", matchingDisplaySet.displaySetInstanceUID);
+          if (matchingDisplaySet.displaySetInstanceUID !== displaySetInstanceUID) {
```

Comment:
So I guess this happens when the availalbeDisplaySets cannot find the DSUID, and it is matched by the SeriesUID, right? But, if we have used DS to match in the HP, there is always a garantee that it exists (since we matched and sorted them out).

Reply:
Yes, will delete that code.